### ISSUE TITLE
InvalidArgumentError: Graph execution error

### ISSUE THREAD

**Issue by niquejoe** [nicknochnack/LipNet#6]
I was trying out the new code and running the training and model fitting, however, everytime I complete 1 epoch I get this error. Any idea?

InvalidArgumentError: Graph execution error:

ValueError: Attempt to convert a value (None) with an unsupported type (<class 'NoneType'>) to a Tensor.
Traceback (most recent call last):

**Comment by Mochoye** [comment]
Facing Similar Issue

**Comment by Anirudh-UG** [comment]
Hey so i am running Python 3.9 on a ubuntu 22.04 system...

I got a similar error when running the code in my local environment.

I couldnt find the exact reason for this error but when degrading my tensorflow version from 2.12.* to 2.11.* the issue got resolved.

Kindly check if it works for you too :)

**Comment by dangarharshil11** [comment]
@nicknochnack I am getting same error can you help?

**Comment by niquejoe** [comment]
I was able to solve it by rewriting some code to work on the new version of the libraries,.

**Comment by mhani6** [comment]
if you'are working on colab it may be the problem, i had the same problem on colab but not on VS code

**Comment by AdamayB** [comment]
Has anyone come up with a solution for this? I am on colab and it doesn't work there.

**Comment by AdamayB** [comment]
> I was able to solve it by rewriting some code to work on the new version of the libraries,.

Can you please provide this code? @niquejoe

**Comment by durgaprasad-04** [comment]
I was trying out the  code and running the training set, however, everytime I complete 1 epoch I get this error.                       
InvalidArgumentError: Graph execution error:  please resolve this.

**Comment by durgaprasad-04** [comment]
> I was trying out the new code and running the training and model fitting, however, everytime I complete 1 epoch I get this error. Any idea?
> 
> InvalidArgumentError: Graph execution error:
> 
> ValueError: Attempt to convert a value (None) with an unsupported type (<class 'NoneType'>) to a Tensor. Traceback (most recent call last):

Did u solve that error ?

**Comment by niquejoe** [comment]
You can check on the version of the libraries you are using, for them to work you must have the same version in the author's code, if not you need to adjust the codes based on the version of the library you are using. In my case, I use it on another code I'm making and adjust the code accordingly.

**Comment by AdamayB** [comment]
> You can check on the version of the libraries you are using, for them to work you must have the same version in the author's code, if not you need to adjust the codes based on the version of the library you are using. In my case, I use it on another code I'm making and adjust the code accordingly.

Yes, that worked for me. Basically, I just installed the libraries without looking at the versions(I was a newbie back then). Then I changed the versions and restarted the entire thing, and it worked!!!

**Comment by WaspsHornets** [comment]
Hallo all, got the same graph execution error, after the first epoch. 
I am not sure if i understand the underlying problem correctly (I am a newbie to ML in TF), but I got it fixed for me. 
So there are some things I changed:
In the load_video(path) function is a Problem with the rgb_to_grayscale() function. I changed the code to this to fix it:
```
`
def load_video(path):
    frames = []
    cap = cv2.VideoCapture(path)
    while cap.isOpened():
        ret, frame = cap.read()
        if ret:
            frame = tf.image.rgb_to_grayscale(frame)  # Convert to grayscale
            frame = tf.image.resize(frame, (128, 128))  # Resize the frame
            frame = tf.reshape(frame, (frame.shape[0], frame.shape[1], 1))
            frames.append(frame)
        else:
            break
    cap.release()
    return tf.stack(frames, axis=0) 
`
```

After that, because we changed the shape of the models input, we have to change it in the model accordingly or this will give us an error. I changed it like this:

```
`model = Sequential()
model.add(Conv3D(128, 3, input_shape=(75,128,128,1), padding='same'))
model.add(Activation('relu'))
model.add(MaxPool3D((1,2,2)))

model.add(Conv3D(256, 3, padding='same'))
model.add(Activation('relu'))
model.add(MaxPool3D((1,2,2)))

model.add(Conv3D(75, 3, padding='same'))
model.add(Activation('relu'))
model.add(MaxPool3D((1,2,2)))

model.add(TimeDistributed(Flatten()))

model.add(Bidirectional(LSTM(128, kernel_initializer='Orthogonal', return_sequences=True)))
model.add(Dropout(.5))

model.add(Bidirectional(LSTM(128, kernel_initializer='Orthogonal', return_sequences=True)))
model.add(Dropout(.5))

model.add(Dense(char_to_num.vocabulary_size()+1, kernel_initializer='he_normal', activation='softmax'))`

```
I hope this fixes the problem for you all!

**Comment by durgaprasad-04** [comment]
> > You can check on the version of the libraries you are using, for them to work you must have the same version in the author's code, if not you need to adjust the codes based on the version of the library you are using. In my case, I use it on another code I'm making and adjust the code accordingly.
> 
> Yes, that worked for me. Basically, I just installed the libraries without looking at the versions(I was a newbie back then). Then I changed the versions and restarted the entire thing, and it worked!!!

could you plz provide the code ??
